src/radio-group.ts

```diff
@@ -554,202 +508,190 @@ export default class RadioGroup extends LitElement implements FormControl {
-        case 'Enter': {
-          this.form?.requestSubmit();
-          break;
-        }
-        case 'ArrowUp':
-        case 'ArrowLeft': {
-          event.preventDefault();
-
-          const previousEnabledRadio = this.#radioElements
-            .slice(0, this.#radioElements.indexOf(event.target))
-            .findLast((radio) => !radio.disabled);
-
-          const lastEnabledRadio = this.#radioElements.findLast(
-            (radio) => !radio.disabled,
-          );
-
-          if (previousEnabledRadio && previousEnabledRadio !== event.target) {
-            this.#uncheckRadio(event.target);
-            this.#checkRadio(previousEnabledRadio);
-          } else if (lastEnabledRadio && lastEnabledRadio !== event.target) {
-            this.#uncheckRadio(event.target);
-            this.#checkRadio(lastEnabledRadio);
-          }
+    if (event.target?.disabled) {
+      return;
+    }
 
-          break;
-        }
-        case 'ArrowDown':
-        case 'ArrowRight': {
-          event.preventDefault();
-
-          const nextEnabledRadio = this.#radioElements.find((radio, index) => {
-            return (
-              !radio.disabled &&
-              event.target instanceof RadioGroupRadio &&
-              index > this.#radioElements.indexOf(event.target)
-            );
-          });
-
-          const firstEnabledRadio = this.#radioElements.find(
-            (radio) => !radio.disabled,
-          );
-
-          if (nextEnabledRadio && nextEnabledRadio !== event.target) {
-            this.#uncheckRadio(event.target);
-            this.#checkRadio(nextEnabledRadio);
-          } else if (firstEnabledRadio && firstEnabledRadio !== event.target) {
-            this.#uncheckRadio(event.target);
-            this.#checkRadio(firstEnabledRadio);
-          }
+    if (event.key === 'Enter') {
+      this.form?.requestSubmit();
 
-          break;
-        }
-        case ' ': {
-          event.preventDefault();
+      return;
+    }
 
-          if (!event.target.disabled && !event.target.checked) {
-            this.#checkRadio(event.target);
+    if (['ArrowUp', 'ArrowLeft'].includes(event.key)) {
+      event.preventDefault(); // Prevent page scroll
 
-            for (const radio of this.#radioElements) {
-              if (radio !== event.target) {
-                this.#uncheckRadio(radio);
-              }
-            }
-          }
+      const previousEnabledRadio = this.#radioElements
+        .slice(0, this.#radioElements.indexOf(event.target))
+        .findLast((radio) => !radio.disabled);
 
-          break;
-        }
+      const lastEnabledRadio = this.#radioElements.findLast(
+        (radio) => !radio.disabled,
+      );
+
+      if (previousEnabledRadio && previousEnabledRadio !== event.target) {
+        event.target.tabIndex = -1;
+        event.target.checked = false;
+
+        previousEnabledRadio.checked = true;
+        previousEnabledRadio.tabIndex = 0;
+        previousEnabledRadio.focus();
```

Comment:
`#checkRadio()` was previously used to set these properties and call `focus()`. But it was used inconsistently and, by setting `tabIndex` and focusing, did much more than check a radio.